service/test/controllers/LearnerControllerTest.java

```diff
@@ -127,12 +199,88 @@ public void testUpdateContentStateFailureForMissingCourseId() {
     requestMap.put(JsonKey.REQUEST, innerMap);
     String data = mapToJson(requestMap);
     JsonNode json = Json.parse(data);
-    RequestBuilder req =
-            new RequestBuilder().bodyJson(json).uri("/v1/content/state/update").method("PATCH");
-    req.headers(new Http.Headers(headerMap));
-    Result result = route(app, req);
-    String response = Helpers.contentAsString(result);
-    System.out.println(response);
-    assertEquals(ResponseCode.CLIENT_ERROR.getResponseCode(), result.status());
+    Http.RequestBuilder req =
+            new Http.RequestBuilder()
+                    .uri(CONTENT_STATE_UPDATE_URL)
+                    .bodyJson(json)
+                    .method("PATCH");
+    Result result = Helpers.route(application, req);
+    Assert.assertEquals( 400, result.status());
+
+  }
+
+  @Test
+  public void testUpdateContentStateFailureWithoutContentId() {
+    Map<String, Object> requestMap = new HashMap<>();
+    Map<String, Object> innerMap = new HashMap<>();
+    List<Object> list = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.CONTENT_ID, null);
+    map.put(JsonKey.STATUS, "Active");
+    map.put(JsonKey.LAST_UPDATED_TIME, "2017-12-18 10:47:30:707+0530");
+    list.add(map);
+    innerMap.put("contents", list);
+    innerMap.put("courseId", COURSE_ID);
+    requestMap.put(JsonKey.REQUEST, innerMap);
+    String data = mapToJson(requestMap);
+    JsonNode json = Json.parse(data);
+    Http.RequestBuilder req =
+            new Http.RequestBuilder()
+                    .uri(CONTENT_STATE_UPDATE_URL)
+                    .bodyJson(json)
+                    .method("PATCH");
+    Result result = Helpers.route(application, req);
+    Assert.assertEquals( 400, result.status());
+
+  }
+
+  @Test
+  public void testUpdateContentStateFailureWithoutStatus() {
+    Map<String, Object> requestMap = new HashMap<>();
+    Map<String, Object> innerMap = new HashMap<>();
+    List<Object> list = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.CONTENT_ID, CONTENT_ID);
+    map.put(JsonKey.STATUS, null);
+    map.put(JsonKey.LAST_UPDATED_TIME, "2017-12-18 10:47:30:707+0530");
+    list.add(map);
+    innerMap.put("contents", list);
+    innerMap.put("courseId", COURSE_ID);
+    requestMap.put(JsonKey.REQUEST, innerMap);
+    String data = mapToJson(requestMap);
+    JsonNode json = Json.parse(data);
```

Comment:
try to refactor the body json create duplicate code to a separate method with argument which allow different changes